### DISCUSSION TITLE
Can't create glfw window

### DISCUSSION THREAD

**Discussion by MohammedBurkhsh** [MuhammadSulaiman001/opengl-lab#15] [Q&A]
كل ما شغل مشروع  يلي نزلتهن من الريبو عم يعطيني هاد الخطأ 
صرلي أسبوع عم عالج فيه وما طلع معي شي , شو الحل
Failed to create GLFW window

C:\Users\MOHAMMED-BURKHUSH\Desktop\opensource\LearnOpenGL\bin\1.getting_started\Debug\1.getting_started__1.1.hello_window.exe (process 9944) exited with code -1.
To automatically close the console when debugging stops, enable Tools->Options->Debugging->Automatically close the console when debugging stops.
Press any key to close this window . . .

**Comment by Red1C3** [comment]
فيك تبعت الاستدعاء اللي عم ينشأ النافذة و المتغيرات المستعملها فيه اذا متسعمل؟؟

**Reply by MuhammadSulaiman001** [reply]
https://github.com/JoeyDeVries/LearnOpenGL/blob/3e94252892660902bef62068c35253cbe3464c9b/src/1.getting_started/1.1.hello_window/hello_window.cpp#L28

**Reply by Red1C3** [reply]
ممكن يكون ما معرف كرت الشاشة :/

**Reply by MohammedBurkhsh** [reply]
تأكدت من كرت الشاشة عندي بيدعم إصدار opengl 4.6

**Comment by MuhammadSulaiman001** [comment]
Please try the following..

1. Add this code after `glfwInit();`

```
glfwSetErrorCallback([](int i, const char* msg)
{
	std::cout << "Error Context:\n"
		<< "Code = " << i << std::endl
		<< "Message = " << msg << std::endl;
});
```

Now, run the project, copy the text in the console and post it in here please.

2. Make sure the configuration is 'Debug', not 'Release', and try running the project again.. 

![image](https://github.com/MuhammadSulaiman001/opengl-lab/assets/36730723/0d0be9c3-8e92-4abc-b771-0cebfd7caa25)

Try changing the platform to x86, then run (if does not work, reset it back to x64 please).

3. Download *Windows pre-compiled binaries* from [here](https://www.glfw.org/download), then try replacing 'glfw3.lib' file in 'LearOpenGL\lib' folder with 'glfw3.lib' from 'lib-vc20xx' (try 2022, 2019, etc..)

4. If you've any other C++ compiler (ex. MinGW), uninstall it, remove everything in 'build' directory, rebuild the solution again (`cmake ..` step).

5. Attach 'glfw.lib' in this discussion please.

**Reply by MohammedBurkhsh** [reply]
يعطيك العافية استاذ محمد.
اتبعت الخطوات كلها يلي عطيتني ياها وما زبط
حتى رجعت حذفت vs 2022  ورجعت ثبتو 😅

**Comment by MuhammadSulaiman001** [comment]
Follow up..
@MohammedBurkhsh Have you clone this repo (i.e opengl-lab) and build/run LabDemos project? please let me know!

**Reply by MohammedBurkhsh** [reply]
لا هدول ما عملتهن clone 
عملت بس LearnOpenGl

**Reply by MuhammadSulaiman001** [reply]
You should have.
If first sample worked.. you can use glfw3.lib from `OpenGL.SharedModule` in this repo!

**Reply by MohammedBurkhsh** [reply]
السلام عليكم
إستاذ بالنسبة إلي ما زبطت البئية أبدا على vc2022 
متل مو بعتلك فوق كلو أخطاء وأي كود عم جيبو نفس الخطأ دغري عم يروح failed to create glfw window 
مشان هيك مسموح استخدم محرر أكواد غير vc202
بس لمشي الوظيفة 😅 
وشكرآ

**Reply by MuhammadSulaiman001** [reply]
وعليكم السلام ورحمة الله،
اكيد مسموح، بس المفروض المشكلة ما الها علاقة بال visual studio.. انت بدك تستخدم كومبايلر تاني؟ (انا ذكرت بأحد الخطوات انو اذا كومبايلر تاني لازم تشيله مشان للcmake تستخدم كومبايلر مايكروسوفت)..

**Reply by MohammedBurkhsh** [reply]
أي تمام ، يعطيك العافية إستاذ محمد 🙆

**Reply by MuhammadSulaiman001** [reply]
الله يعافيك, اذا انحلت المشكلة معك اكتب شو كان الحل مشان نستفيد واذا حدا واجهته المشكلة كمان يعرف كيف انحلت معك
